blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -364,9 +356,17 @@ public MetricData getDatapointsForRange(Locator locator, Range range, Granularit
         for (Locator locator : locators) {
             try {
                 RollupType rollupType = RollupType.fromString((String)
-                        metaCache.get(locator, MetricMetadata.ROLLUP_TYPE.name().toLowerCase()));
-                DataType dataType = new DataType((String)
-                        metaCache.get(locator, MetricMetadata.TYPE.name().toLowerCase()));
+                            metaCache.get(locator, MetricMetadata.ROLLUP_TYPE.name().toLowerCase()));
+                if (rollupType == null) {
+                    rollupType = RollupType.BF_BASIC;
+                }
+                //TODO: If we stop processing string and boolean, we can always hardcode this to numeric
+                DataType dataType = getDataType(locator, MetricMetadata.TYPE.name().toLowerCase());
+
+                if (dataType == null) {
+                    dataType = DataType.INT;
```

Comment:
2:22 PM <chinmay_g_> tilogaat: this seems to be really hacky https://github.com/rackerlabs/blueflood/pull/412/files#diff-d3454fc0815b5e706f9fabec298fdc48R367
2:23 PM <tilogaat> chinmay_g_: you mean not returning the locator if number of datapoints is 0
2:23 PM <tilogaat> ?
2:23 PM <chinmay_g_> because data we are going to retrieve can be Long, Double or BigInt as well
2:23 PM <chinmay_g_> no, sec, I will explain
2:24 PM <chinmay_g_> dominiclobue: can you make the rollup thread settings ASAP, we are still falling back on rollup work.....
2:25 PM <tilogaat> chinmay_g_: yes I agree, but it does not really matter because the serializers are self describin
2:25 PM <tilogaat> they dont depend on the datatype to return the serializer
2:25 PM <tilogaat> datatype is just used to return the serializer-deserializer for string or bool
2:26 PM <chinmay_g_> yeah, should we refactor the code then to have just have an umbrella Numeric type?
2:26 PM <tilogaat> maybe i can add tests to show that even if data type is int, we return the correct serializer
2:26 PM <tilogaat> chinmay_g_: yes, we can 
2:26 PM <chinmay_g_> like there will be just 3 types 1) Numeric 3) String 3) Boolean
2:26 PM <chinmay_g_> cool
2:27 PM <chinmay_g_> tilogaat: this is turning into more contrived task than we imagined, no?
2:27 PM <chinmay_g_> do you want me to add some commits?
2:27 PM <tilogaat> chinmay_g_: yes .. i was thinking of just making another PR and remove the rollup type in that 
2:27 PM <tilogaat> so that its easier to deploy and review 
2:27 PM <tilogaat> then anotehr for units and anotehr for type
2:27 PM <tilogaat> basically cherry pick the commits but deploy and review them separately
2:28 PM <chinmay_g_> tilogaat: yea, that is a good idea...
2:28 PM <tilogaat> ok i will do that